commands/permissions.go

```diff
@@ -71,29 +126,38 @@ func addPermissionsCmdF(c client.Client, cmd *cobra.Command, args []string) erro
 	return nil
 }
 
-func removePermission(permissions []string, permission string) []string {
-	newPermissions := []string{}
-	for _, p := range permissions {
-		if p != permission {
-			newPermissions = append(newPermissions, p)
-		}
-	}
-	return newPermissions
-}
-
 func removePermissionsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	role, response := c.GetRoleByName(args[0])
 	if response.Error != nil {
 		return response.Error
 	}
 
-	newPermissions := role.Permissions
-	for _, arg := range args[1:] {
-		newPermissions = removePermission(newPermissions, arg)
+	newPermissionSet := role.Permissions
+	for _, permissionID := range args[1:] {
+		newPermissionSet = removeFromStringSlice(newPermissionSet, permissionID)
+	}
+
+	if ok, _ := cmd.Flags().GetBool(flagAncillaryPermissions); ok {
```

Comment:
nit: I'd prefer to use something like `ancillary` here instead of the `ok`, as it confused me a little at first with the next two map lookups:

```suggestion
	if ancillary, _ := cmd.Flags().GetBool(flagAncillaryPermissions); ancillary {
```